osf/management/commands/sync_datacite_doi_metadata.py

```diff
@@ -10,31 +10,38 @@
 
 logger = logging.getLogger(__name__)
 
-def sync_datacite_doi_metadata(dry_run=True, retries=4):
+def retry(func, retries=4):
+    for i in range(retries):
+        try:
+            time.sleep(0.3)
+            return func('doi')
+        except DataCiteForbiddenError as e:
+            if i < 1:
+                raise e
+            time.sleep(10)
+
+def sync_datacite_doi_metadata(dry_run=True, batch_size=100):
     content_type = ContentType.objects.get_for_model(Registration)
     reg_ids = Identifier.objects.filter(category='doi', content_type=content_type, deleted__isnull=True).values_list(
         'object_id',
         flat=True
     )
 
-    registrations = Registration.objects.exclude(id__in=reg_ids)
-    logger.info(f'{registrations.count()} registrations to mint')
+    registrations = Registration.objects.exclude(
+        id__in=reg_ids,
+        deleted__isnull=False,
+        moderation_state='withdrawn'
+    )[:batch_size or 1 - 1]
```

Comment:
Why the conditional logic around `batch_size` here, and if you want to default to everything in the case of a `False`-y `batch_size`, shouldn't this be `[:batch_size or -1]`?

Reply:
I must have been drinking heavily when I wrote this.